app/scripts/controllers/visualizerModal.js

```diff
@@ -1,29 +1,37 @@
 'use strict';
 
 angular.module('openhimConsoleApp')
-  .controller('VisualizerModalCtrl', function ($http, $scope, $modalInstance, $timeout, Api, Notify, Alerting, visualizers) {
-
-
-    /***************************************************/
-    /**         Initial page load functions           **/
-    /***************************************************/
+  .controller('VisualizerModalCtrl', function ($http, $scope, $modalInstance, $timeout, Api, Notify, Alerting, visualizers, visualizer, duplicate) {
 
     $scope.ngError = {};
     $scope.validationRequiredMsg = 'This field is required.';
 
     // get/set the users scope whether new or update
-    $scope.update = false;
-
-    // create visualizer settings properties
-    $scope.visualizer = {};
-
-    // load default visualizer config for new user
-    $http.get('config/visualizer.json').success(function( visualizerConfig ) {
-      angular.extend( $scope.visualizer, angular.copy( visualizerConfig ) );
-    });
+    if (visualizer) {
+      $scope.update = true;
+      // make a copy of the object so that the original doesn't get changed until we save
+      $scope.visualizer = JSON.parse(angular.toJson(visualizer));
```

Comment:
I was getting call stack exceeded with `angular.copy`, I think now that I'm using `angular.toJson` it may work again as it cuts out all the angular specific properties (which seem to have recursive references). I'l give this another go.